File: MoonEngine-Root/Base/src/main/java/com/greentree/engine/moon/base/name/Name.java
package com.greentree.engine.moon.base.name;

import java.util.Objects;

import com.greentree.common.ecs.component.ConstComponent;

public record Name(String value) implements ConstComponent {
	
	
	public Name {
		Objects.requireNonNull(value);
	}
	
	@Override
	public String toString() {
		return "Name [" + value + "]";
	}
	
}

File: MoonEngine/Base/src/main/java/com/greentree/engine/moon/base/name/Name.kt
package com.greentree.engine.moon.base.name

import com.greentree.engine.moon.ecs.component.ConstComponent
import java.util.*

data class Name(val value: String) : ConstComponent {

	fun value() = value

	override fun toString(): String {
		return "Name [$value]"
	}

	init {
		Objects.requireNonNull(value)
	}
}